odoo/models.py

```diff
@@ -1892,6 +1892,43 @@ def name_search(self, name='', args=None, operator='ilike', limit=100) -> list[t
         records = self.search_fetch(domain, ['display_name'], limit=limit)
         return [(record.id, record.display_name) for record in records.sudo()]
 
+    @api.model
+    def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
+        """ _name_search(name='', domain=None, operator='ilike', limit=None, order=None) -> ids
```

Comment:
Creating a new method from scratch will not make it called by magic.  This should ring a bell.
The method `_name_search()` has actually been removed in Odoo 18.0.  Go have a look to `_search_display_name` instead.